src/server/game/Entities/Player/Player.cpp

```diff
@@ -6478,6 +6478,7 @@ void Player::CheckAreaExploreAndOutdoor()
                     XP = uint32(sObjectMgr->GetBaseXP(areaEntry->area_level)*sWorld->getRate(RATE_XP_EXPLORE));
                 }
 
+                XP = uint32(XP * (IsEventActive(sWorld->getIntConfig(CONFIG_RATE_XP_WEEKEND_EVID)) ? sWorld->getRate(RATE_XP_WEEKEND) : 1.0f));
```

Comment:
I'm open to constructive feedback, if you have a better idea :)

On Aug 19, 2016 5:06 AM, "Treeston" <notifications@github.com> wrote:

> In src/server/game/Entities/Player/Player.cpp
> <https://github.com/TrinityCore/TrinityCore/pull/17782#discussion_r75449383>
> :
>
> > @@ -6478,6 +6478,7 @@ void Player::CheckAreaExploreAndOutdoor()
> >                      XP = uint32(sObjectMgr->GetBaseXP(areaEntry->area_level)*sWorld->getRate(RATE_XP_EXPLORE));
> >                  }
> >
> > +                XP = uint32(XP * (IsEventActive(sWorld->getIntConfig(CONFIG_RATE_XP_WEEKEND_EVID)) ? sWorld->getRate(RATE_XP_WEEKEND) : 1.0f));
>
> That's a terrible name for a method.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/TrinityCore/TrinityCore/pull/17782/files/29e3cc30b2c9d32eedd2546a81fd1cc4bfd54441#r75449383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA5XAhrP0654vaTJrss-zZM3UwNpclVrks5qhXIUgaJpZM4JkA4S>
> .
>